returnsrcvr-checker/src/main/java/org/checkerframework/checker/returnsrcvr/ReturnsRcvrAnnotatedTypeFactory.java

```diff
@@ -55,61 +62,37 @@ public Void visitExecutable(AnnotatedTypeMirror.AnnotatedExecutableType t, Void
-      ExecutableElement element = t.getElement();
-
       // skip constructors
-      if (element.getKind().equals(ElementKind.CONSTRUCTOR)) {
-        return false;
-      }
-
-      Element enclosingElement = element.getEnclosingElement();
-
-      boolean inAutoValueBuilder = hasAnnotation(enclosingElement, AutoValue.Builder.class);
-      boolean inLombokBuilder =
-          (hasAnnotationByName(enclosingElement, "lombok.Generated")
-                  || hasAnnotationByName(element, "lombok.Generated"))
-              && enclosingElement.getSimpleName().toString().endsWith("Builder");
-
-      if (!inAutoValueBuilder && !inLombokBuilder) {
-        // see if superclass is an AutoValue Builder, to handle generated code
-        TypeMirror superclass = ((TypeElement) enclosingElement).getSuperclass();
-        // if enclosingType is an interface, the superclass has TypeKind NONE
-        if (!superclass.getKind().equals(TypeKind.NONE)) {
-          // update enclosingElement to be for the superclass for this case
-          enclosingElement = TypesUtils.getTypeElement(superclass);
-          inAutoValueBuilder = enclosingElement.getAnnotation(AutoValue.Builder.class) != null;
+      if (!isConstructor(t)) {
+        // check each supported framework
+        for (FrameworkSupport frameworkSupport : frameworkSupports) {
+          // see if the method in the framework should return this
+          if (frameworkSupport.knownToReturnThis(
+              t, (ReturnsRcvrAnnotatedTypeFactory) this.typeFactory)) {
+            // add @This annotation
+            returnType.replaceAnnotation(THIS_ANNOT);
+            AnnotatedTypeMirror.AnnotatedDeclaredType receiverType = t.getReceiverType();
+            receiverType.replaceAnnotation(THIS_ANNOT);
+            break;
+          }
         }
       }
 
-      if (inAutoValueBuilder || inLombokBuilder) {
-        AnnotatedTypeMirror returnType = t.getReturnType();
-        return returnType != null
-            && enclosingElement.equals(TypesUtils.getTypeElement(returnType.getUnderlyingType()));
-      }
-
-      return false;
+      return super.visitExecutable(t, p);
     }
   }
 
-  private boolean hasAnnotation(Element element, Class<? extends Annotation> annotClass) {
+  private boolean isConstructor(AnnotatedTypeMirror.AnnotatedExecutableType t) {
+    ExecutableElement element = t.getElement();
+    return element.getKind().equals(ElementKind.CONSTRUCTOR);
+  }
+
+  public boolean hasAnnotation(Element element, Class<? extends Annotation> annotClass) {
     return elements.getAllAnnotationMirrors(element).stream()
         .anyMatch(anm -> AnnotationUtils.areSameByClass(anm, annotClass));
   }
 
-  private boolean hasAnnotationByName(Element element, String annotClassName) {
+  public boolean hasAnnotationByName(Element element, String annotClassName) {
     return elements.getAllAnnotationMirrors(element).stream()
```

Comment:
So we only use `elements` to call `getAllAnnotationMirrors()` which handles inherited annotations.  But for the use cases here I don't think we care about inherited annotations.  So, we should be able to replace the call as below and then make the enclosing method `static`:
```suggestion
     return element.getAnnotationMirrors().stream()
```